### DISCUSSION TITLE
Dynamic matrix re-run all jobs causing duplicate jobs to appear

### DISCUSSION THREAD

**Discussion by chingc** [community/community#25466] [Actions]
<p>I’m using a dynamic matrix and I think this is causing duplicates to show up on the left sidebar when I use “Re-run all jobs”.  Here’s an image to illustrate:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/2/324162b3057a81aa8631edc16ca967ef5c718706.png" data-download-href="/uploads/short-url/7azXBDF6q7DM4j0jZGm68X16UaW.png?dl=1" title="" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/471856/181091234-e1a8d5e3-a7fd-413f-9833-3214ff61b767.png" alt="" data-base62-sha1="7azXBDF6q7DM4j0jZGm68X16UaW" width="287" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/3/2/324162b3057a81aa8631edc16ca967ef5c718706_2_287x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/3/2/324162b3057a81aa8631edc16ca967ef5c718706_2_430x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/3/2/324162b3057a81aa8631edc16ca967ef5c718706_2_574x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/2/324162b3057a81aa8631edc16ca967ef5c718706_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename"></span><span class="informations">652×1134 37.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>As you can see, “Group 1” was cancelled by fail-fast.  When I used “Re-run all jobs” it caused a duplicate set of groups to appear.  How can I avoid this?</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/chingc">@chingc</a>,</p>
<p>This is a known UI issue similar to that reported in this earlier topic:</p><aside class="quote quote-modified" data-post="1" data-topic="135997">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/specific-workflow-runs-keeps-failing-with-an-internal-error-on-macos/135997">Specific workflow runs keeps failing with an internal error on macOS</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    In these 2 runs, some macOS jobs keep failing with an “internal error” no matter how many times I re-run them: 

 


I have tried a few days/weeks ago, as well as right now, same result. 
The jobs don’t even start, they just go straight to showing “This check has failed” with a big “X” in the middle of the job log. What’s going on?
  </blockquote>
</aside>

<p>I had created an issue ticket (<a href="https://github.com/actions/runner/issues/742" rel="noopener nofollow ugc">actions/runner#742</a>) to report the problem to the appropriate engineering team for further investigation and evaluation.</p>
<p>you can follow the issue ticket and add your comments to it.</p>

**Comment by chingc** [comment, marked as answer]
<p>I was able to fix this issue and I think I know what’s going on.</p>
<p>In my original dynamic matrix I had group index and group files.  The way my workflow was configured, the files were always shuffled before using <code>set-output</code>… so even if I had the same number of groups they would refer to a different set of files.  I think this is what caused the duplicates to appear.</p>
<p>The fix was to separate them into two dynamic matrixes.  I use the group index matrix as the matrix for my dependent jobs.  Then I would set the group files matrix (which is a json string) as an environment variable.  Finally, I use jq and the group index to pluck out the correct group files.</p>
<p>Hope that makes sense.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/chingc">@chingc</a>,</p>
<p>Glad that the problem has been solved.<br>
And thanks for sharing your experience, this may be helpful to other people who have the similar problems.</p>
